cmake/modules/hunter_check_toolchain_for_define.cmake

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2016 NeroBurner
+# All rights reserved.
+
+include(hunter_test_string_not_empty)
+
+function(hunter_check_toolchain_for_define definition output_var)
+  # the toolchain path must be set
+  hunter_test_string_not_empty("${HUNTER_TOOLCHAIN_ID_PATH}")
+  set(TOOLCHAIN_INFO_FILE "${HUNTER_TOOLCHAIN_ID_PATH}/toolchain.info")
+
+  # read toolchain.info file into memory, each line a entry in a list
+  file(STRINGS "${TOOLCHAIN_INFO_FILE}" toolchain_contents)
+
+  # clean ouptut variable
+  set(definition_found)
+
+  foreach(line ${toolchain_contents})
+    # find the specified define
+    string(REGEX MATCH "^#define ${definition} .*$" match "${line}")
+    if(match)
+      # extract the definition
+      string(REGEX REPLACE "^#define ${definition} " "" definition_found "${match}")
```

Comment:
> If I just return a boolean (1/0), then what should #define B 0 return?

Will the code inside `#if defined(A)` be evaluated?

```cpp
#define A 0
#if defined(A)
  // some code
#endif
```

As far as I know yes. Hence we should return YES.

Just like I said I was thinking about function that will just return YES if **defined** and NO if **not defined**, nothing about the real value. If you want value then you should introduce two output variables, like:
```cmake
hunter_check_toolchain_definition(NAME A DEFINED a_defined VALUE a_value)
```
in this case `#defined A` will return `a_defined=YES`/`a_value=""`, `#defined A 0` will return `a_defined=YES`/`a_value=0`, etc.